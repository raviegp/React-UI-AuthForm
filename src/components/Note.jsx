import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import DeleteIcon from '@material-ui/icons/Delete';
import EditIcon from '@material-ui/icons/Edit';
import { Card, CardHeader, CardContent, CardActions } from '@material-ui/core';
import { Link } from 'react-router-dom';
import highlightSearchTerm from '../utils/highlight-search-term';

const styles = theme => ({
    root: {
        maxWidth: theme.spacing.unit * 50,
        minWidth: theme.spacing.unit * 40,
        [theme.breakpoints.up('sm')]: {
            maxWidth: theme.spacing.unit * 30,
            minWidth: theme.spacing.unit * 20,
        }, 
        [theme.breakpoints.up('md')]: {
            maxWidth: theme.spacing.unit * 50,
            minWidth: theme.spacing.unit * 40
        }
    },
    card: {
        minWidth: 275,
    },
    pos: {
        marginBottom: 12,
    },
    deleteIcon: {
        justifyContent: 'flex-end', 
    },
});

const Note = (props) => {
    const { classes, note, handleRemoveNote, searchTerm } = props;
    return (
        <Card className={classes.root}>
            <CardHeader 
                action={
                    <Link to={`/edit-note/${note.id}`}>
                        <IconButton>
                            <EditIcon />
                        </IconButton>
                    </Link>
                }
                title={searchTerm ? highlightSearchTerm(note.noteTitle, searchTerm) : note.noteTitle}
            />
            <CardContent>
                <Typography
                    variant="h6"
                    component="h2"
                >   
                    { searchTerm ? highlightSearchTerm(note.noteDescription, searchTerm) : note.noteDescription }
                </Typography>
            </CardContent>
            <CardActions className={classes.deleteIcon}>
                <IconButton onClick={handleRemoveNote.bind(null, note.id)}>
                    <DeleteIcon />
                </IconButton>
            </CardActions>
        </Card>
    );
}

Note.propTypes = {
    classes: PropTypes.object.isRequired,
    note: PropTypes.object.isRequired,
    handleRemoveNote: PropTypes.func.isRequired,
    searchTerm: PropTypes.string,
};

export default withStyles(styles)(Note);