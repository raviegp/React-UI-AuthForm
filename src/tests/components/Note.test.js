import Note from '../../components/Note';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import notes from '../fixtures/notes';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

let wrapper, handleRemoveNote
beforeEach(() => {
    handleRemoveNote = jest.fn();
    wrapper = shallow(
        <Note
            note={notes[0]}
            handleRemoveNote={handleRemoveNote}
        />
    );
});

test('should render Note component correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

