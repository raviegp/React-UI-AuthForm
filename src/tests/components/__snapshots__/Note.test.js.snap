// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Note component correctly 1`] = `
<Note
  classes={
    Object {
      "card": "Note-card-2",
      "deleteIcon": "Note-deleteIcon-4",
      "pos": "Note-pos-3",
      "root": "Note-root-1",
    }
  }
  handleRemoveNote={[MockFunction]}
  note={
    Object {
      "id": 1,
      "noteDescription": "Note 1 description",
      "noteTitle": "Note 1",
    }
  }
/>
`;
